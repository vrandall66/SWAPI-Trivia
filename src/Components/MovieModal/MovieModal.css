.MovieModalContainer {
  display: flex;
  flex-direction: column;
}

.MovieModal__btn--close {
  height: 42px;
  width: 42px;
}