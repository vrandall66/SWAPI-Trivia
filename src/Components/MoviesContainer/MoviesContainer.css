.MoviesContainer {
  align-items: flex-end;
  color: #ffe81f;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  height: 100vh;
}

.MoviesContainer-planets {
  align-content: space-around;
  justify-content: space-evenly;
  display: flex;
  width: 60%;
  height: 80vh;
  flex-wrap: wrap;
}

.MoviesContainer-loading {
  height: 25%;
  width: 25%;
}
