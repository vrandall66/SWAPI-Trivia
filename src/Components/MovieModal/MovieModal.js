import React from "react";
import { Link } from "react-router-dom";
import "./MovieModal.css";
import lightsaberClose from "../../images/LightsaberCross.png";

const MovieModal = ({ currentMovie, updatePlanetModalState }) => {
  const { title, id, releaseDate, url } = currentMovie;
  return (
    <div className="MovieModalContainer">
      <div className='MovieModalContainer__div'>
        <input
          type="image"
          src={lightsaberClose}
          alt="Crossed lightsabers - close"
          className="MovieModal__btn--close"
          onClick={updatePlanetModalState}
        />
      </div>
      <h1 className="MovieModal__h1">{title}</h1>
      <h3 className="MovieModal__h3">Episode: {id}</h3>
      <h5 className="MovieModal__h5">Release Date:<br/> {releaseDate}</h5>
      <Link to={`/movies/${id}`}>
        <button
          type="button"
          className="MovieModal__btn--characters"
          onClick={updatePlanetModalState}
        >
          View Characters
        </button>
      </Link>
    </div>
  );
};

export default MovieModal;
