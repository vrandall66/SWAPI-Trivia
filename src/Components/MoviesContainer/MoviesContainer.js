import React from "react";
import MoviePlanet from "../MoviePlanet/MoviePlanet";
import "./MoviesContainer.css";

const MoviesContainer = ({ movies, updatePlanetModalState, reactModal, updateCurrentMovie }) => {
  console.log(movies);
  const sortMovies = movies => {
    return movies.sort((a, b) => {
      return a.id - b.id;
    });
  };

  const createMovieObjects = () => {
    const planets = movies.map(movie => {
      return {
        title: movie.title,
        id: movie.episode_id,
        releaseDate: movie.release_date,
        url: movie.url
      };
    });
    return sortMovies(planets);
  };

  const moviePlanets = createMovieObjects().map((movie, index) => {
    return (
      <MoviePlanet
        movie={movie}
        key={index}
        updatePlanetModalState={updatePlanetModalState}
        updateCurrentMovie={updateCurrentMovie}
      />
    );
  });

  return (
    <div className="MoviesContainer">
      <h1>Episodes</h1>
      <div className="MoviesContainer-planets">{moviePlanets}</div>
      {reactModal}
    </div>
  );
};

export default MoviesContainer;
