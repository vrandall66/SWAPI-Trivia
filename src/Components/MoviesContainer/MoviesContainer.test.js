import React from "react";
import { shallow } from "enzyme";
import MoviesContainer from "./MoviesContainer";
import MovieModal from "../MovieModal/MovieModal";
import ReactModal from "react-modal";

describe("MoviesContainer", () => {
  let wrapper;
  let allMovies = [];
  let updatePlanetModalState = jest.fn();
  let updateCurrentMovie = jest.fn();
  let resetMovieState = jest.fn();
  let showPlanetModal = false;
  let currentMovie = {};

  beforeEach(() => {
    wrapper = shallow(
      <MoviesContainer
        movies={allMovies}
        updatePlanetModalState={updatePlanetModalState}
        updateCurrentMovie={updateCurrentMovie}
        className="MoviesContainer"
        resetMovieState={resetMovieState}
        reactModal={
          <ReactModal
            isOpen={showPlanetModal}
            style={{ overlay: {}, content: {} }}
          >
            <MovieModal
              currentMovie={currentMovie}
              updatePlanetModalState={updatePlanetModalState}
            />
          </ReactModal>
        }
      />
    );
  });

  it("should match the snapshot()", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
