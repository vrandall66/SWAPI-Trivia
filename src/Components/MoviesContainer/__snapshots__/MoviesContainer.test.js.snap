// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match the snapshot() 1`] = `
<div
  className="MoviesContainer"
>
  <h1
    className="MoviesContainer__h1--Episodes"
  >
    Episodes
  </h1>
  <div
    className="MoviesContainer-planets"
  />
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {},
        "overlay": Object {},
      }
    }
  >
    <MovieModal
      currentMovie={Object {}}
      updatePlanetModalState={[MockFunction]}
    />
  </Modal>
</div>
`;
